extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosExporter.java

```diff
@@ -0,0 +1,170 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.smugmug.photos;
+
+import static org.dataportabilityproject.transfer.smugmug.photos.SmugMugInterface.ALBUMS_KEY;
+import static org.dataportabilityproject.transfer.smugmug.photos.SmugMugInterface.USER_URL;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult.ResultType;
+import org.dataportabilityproject.spi.transfer.provider.Exporter;
+import org.dataportabilityproject.spi.transfer.types.ContinuationData;
+import org.dataportabilityproject.spi.transfer.types.ExportInformation;
+import org.dataportabilityproject.spi.transfer.types.IdOnlyContainerResource;
+import org.dataportabilityproject.spi.transfer.types.PaginationData;
+import org.dataportabilityproject.spi.transfer.types.StringPaginationToken;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbum;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbumImage;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbumInfoResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugAlbumsResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugResponse;
+import org.dataportabilityproject.transfer.smugmug.photos.model.SmugMugUserResponse;
+import org.dataportabilityproject.types.transfer.auth.AuthData;
+import org.dataportabilityproject.types.transfer.models.photos.PhotoAlbum;
+import org.dataportabilityproject.types.transfer.models.photos.PhotoModel;
+import org.dataportabilityproject.types.transfer.models.photos.PhotosContainerResource;
+
+public class SmugMugPhotosExporter implements Exporter<AuthData, PhotosContainerResource> {
+
+  // TODO(olsona): distinguish between album page tokens and photo page tokens
+
+  static final String ALBUM_URL_FORMATTER = "/api/v2/album/%s!images";
+  private SmugMugInterface smugMugInterface;
+
+  public SmugMugPhotosExporter() {
+    // TODO(olsona)
+    this.smugMugInterface = null;
+  }
+
+  @VisibleForTesting
+  SmugMugPhotosExporter(SmugMugInterface smugMugInterface) {
+    this.smugMugInterface = smugMugInterface;
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData) {
+    return exportAlbums(Optional.empty());
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData,
+      ExportInformation exportInformation) {
+    return null;
+  }
+
+  private ExportResult<PhotosContainerResource> exportAlbums(
+      Optional<PaginationData> paginationData) {
+    try {
```

Comment:
can we wrap the try/catch around just the part that throws this exception (if its only a few lines long) - would make this easier to read imo

Reply:
Done.